src/dataviews/dataview-model.js

```diff
@@ -0,0 +1,121 @@
+var cdb = require('cartodb.js');
+
+/**
+ * Default dataview model
+ */
+module.exports = cdb.core.Model.extend({
+  defaults: {
+    url: '',
+    data: [],
+    columns: [],
+    sync: true,
+    bbox: true,
+    collapsed: false
+  },
+
+  url: function () {
+    return this.get('url') + '?bbox=' + this.get('boundingBox');
+  },
+
+  initialize: function (attrs, opts) {
+    attrs = attrs || {};
+    opts = opts || {};
+
+    if (!attrs.id) {
+      this.set('id', attrs.type + '-' + this.cid);
```

Comment:
Widgets defined in a namedmap (see example above) will have ids pre-defined, otherwise one is generated from the model's cid (only valid for the lifecycle of the webpage though).